lib/watcher.js

```diff
@@ -113,6 +126,30 @@ Watcher.prototype.updateTestDependencies = function (file, sources) {
 	}
 };
 
+Watcher.prototype.trackExclusivity = function (api) {
+	var self = this;
+	api.on('stats', function (stats) {
+		self.updateExlusivity(stats.file, stats.hasExclusive);
+	});
+};
+
+Watcher.prototype.updateExlusivity = function (file, hasExclusive) {
+	var ix = this.exclusiveTests.indexOf(file);
+
+	if (hasExclusive && ix === -1) {
+		this.exclusiveTests.push(file);
+	} else if (!hasExclusive && ix !== -1) {
+		this.exclusiveTests.splice(ix, 1);
```

Comment:
I don't totally get why we're splicing in here at a specific position. Maybe a comment would help.